### PR TITLE
Replace wrench with shelljs

### PR DESCRIPTION
**Pull request by sbauer322**
In this pull request for issue #2237 wrench is swapped out for fs-extra due to the deprecation of wrench. Luckily wrench was only used in two places making this a relatively minor pull request.

There has been no change in functionality with regards to delete and copy within files.cljs as a result.

If desired, we can expand the scope of this PR to update all places we use fs to use fs-extra instead. fs-extra contains fs within it...

### CODE REVIEW

**Comment by sbauer322** [comment]
Please verify everything is hunky dory with regards to npm. After updating package.json, the commands used were:

npm install fs-extra@0.30.0
npm uninstall wrench

And the results committed

**Comment by kenny-evitt** [comment]
@sbauer322 Thanks for this PR! Sorry for taking so long to get around to reviewing it.

But do we even need *fs-extra*? I don't think we do. The *shell* npm package `rm` method seems to handle recursive deletes just fine. The existing upgrade code has been using it for three years. Would you mind testing whether we can simply use the *shell* package methods instead?

**Comment by sbauer322** [comment]
Will do! I had chosen fs-extra simply because that is what wrench
suggested.

On Aug 14, 2016 12:33, "kenny-evitt" <notifications@github.com> wrote:

@sbauer322 <https://github.com/sbauer322> Thanks for this PR! Sorry for
taking so long to get around to reviewing it.

But do we even need *fs-extra*? I don't think we do. The *shell* npm
package rm method seems to handle recursive deletes just fine. The existing
upgrade code has been using it for three years. Would you mind testing
whether we can simply use the *shell* package methods instead?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/LightTable/LightTable/pull/2238#issuecomment-239682869>,
or mute the thread

**Comment by rundis** [comment]
If we can remove npm dependencies I'm all for that. Funnily I happen to use wrench from a couple of my plugins, but that's obviously of no concern to LT. But it should come as a warning in the release notes. Hopefully there aren't many other plugins that use it.

**Comment by kenny-evitt** [comment]
@sbauer322 Thanks!

@rundis You and other plugin authors can pull in *wrench* (or any other Node.js package) independently of LT tho – right? Is the potential issue just that removing it from LT core might break plugins that are using it without including it in their own projects themselves?

**Comment by rundis** [comment]
The latter.
I haven't found anything in the dev docs stating that you shouldn't use the node modules shipping with LT. Maybe it should mention that if you do it's at your own peril, because it's not part of the official "api" for plugins ?

**Comment by kenny-evitt** [comment]
@rundis We should mention that those packages aren't part of the 'official API' – now I'm trying to figure out where best to do that! Any ideas?

**Comment by sbauer322** [comment]
@kenny-evitt my preference would be stating this somewhere within for-committers.md, the wiki, and package.json itself as a comment. It is possible to miss if it is only within a single supporting document.

**Comment by sbauer322** [comment]
I was able to switch `files.cljs` to use [`shelljs`](https://github.com/shelljs/shelljs) instead of `wrench`. Do note, there is a different library(?) named [`shell`](https://github.com/electron/electron/blob/master/docs/api/shell.md) already being used within `files.cljs`.

Sorry for the list, @kenny-evitt, but there are a couple questions now:

- Should we rename what is currently known as shell to something such as `electron-shell` across the project? It can be easily confused with shelljs if you are not aware of both of them... I know I wasn't! :smile: 
- Do you want wrench removed as a dependency within LT or leave it? Alternatively, we could deprecate it for future releases and aim to remove it entirely for 0.9 or 1.0?

**Comment by rundis** [comment]
Tx for your contributions @sbauer322. Much appreciated.
I'd say we go for the following. 
- remove wrench completely
- either rename use of shell to electron-shell as you suggested, or  rename var usagee of shelljs to be named shell-js. Prefer the former.

**Comment by sbauer322** [comment]
Should be all set now @rundis, @kenny-evitt! 

Wrench has been removed, shell has been renamed to electron-shell and shelljs is now shell

**Comment by kenny-evitt** [comment]
@sbauer322 Please rebase the upstream *master* onto this branch.

**Comment by kenny-evitt** [comment]
@sbauer322 @rundis I just tested all of the UI features that use any of the affected vars and everything looks good (or as good as it was before these changes). The only issue I noticed was #2129 but Scott already commented that it's still an issue even with the other related changes.

@sbauer322 I'll merge this after you've rebased the upstream *master* on top of this branch.

**Comment by sbauer322** [comment]
@kenny-evitt: I believe I have done what you asked, but full disclosure - I've never used rebase before. 

The commands used were:

```
git checkout replace-wrench-with-fs-extra
git rebase LightTable/master
```

Could you confirm I did not do something undesired prior? If all is well then merge away!

**Comment by kenny-evitt** [comment]
@sbauer322 I'm not sure what happened. GitHub doesn't show any changes for the 'merge' commit. I think it's okay. Would you squash the three commits as one commit?

**Comment by sbauer322** [comment]
I think we are clear to merge now, @kenny-evitt!

File: src/lt/objs/files.cljs [hunk 1]
```diff
@@ -231,13 +233,13 @@
...
 (defn trash! [path]
-  (.moveItemTotrash shell path))
+  (.moveItemTotrash electron-shell path))
 
 (defn delete!
   "Delete file or directory"
   [path]
   (if (dir? path)
-    (.rmdirSyncRecursive wrench path)
+    (.rm shell "-rf" path)
```
Comment (cldwalker) [inline src/lt/objs/files.cljs]: @sbauer322 Using the force option here is a little concerning. Do you have a specific use case for it? Given that [this option ignores write permissions](https://github.com/shelljs/shelljs/blob/35cef9a4ae2b7321bd8cc341a7528ef015bca304/src/rm.js#L125-L127), any bugs (accidental args) to this fn could lead to harmful deletions for users, especially those running as root

Comment (sbauer322) [inline src/lt/objs/files.cljs]: Nice catch! You are correct, it does not appear force is needed based on some quick testing I just did. I have opened PR #2284 to address this.

**Pull request merged**